src/Modules/Server.cpp

```diff
@@ -753,9 +753,14 @@ static void resetCoopProgress() {
 }
 
 static int g_sendResetDoneAt = -1;
+static void(__rescall *CreateViewModel)(void *, int);
 
 ON_EVENT(SESSION_START) {
 	g_sendResetDoneAt = -1;
+
+	/* spawn viewmodel if it doesnt exist */
+	if (!entityList->GetEntityInfoByClassName("viewmodel") && !engine->demoplayer->IsPlaying())
```

Comment:
This is definitely missing a few checks...
* You shouldn't even attempt this `GetEntityInfoByClassName` if you're orange -- in fact, it should probably be disabled entirely in coop for now, since it won't spawn it for orange (you could alternatively fix that)
* This feature should probably be behind a cvar (maybe disabled by default for now due to legitimacy concerns)